### ISSUE TITLE
Restore from Backup in windows

### ISSUE THREAD

**Issue by tigerclawn82** [restic/restic#360]
Hi, 
   I was testing restic today, i created a snapshot of a directory and it given following errors but completed.

```
error for Workspace\PhpStorm\resume-server\registry-server\node_modules\resume-t
o-html\node_modules\resume-schema\node_modules\z-schema\node_modules\request\nod
e_modules\form-data\node_modules\combined-stream\node_modules\delayed-stream\tes
t\integration\test-delayed-http-upload.js: GetFileAttributesEx D:\Workspace\PhpS
torm\resume-server\registry-server\node_modules\resume-to-html\node_modules\resu
me-schema\node_modules\z-schema\node_modules\request\node_modules\form-data\node
_modules\combined-stream\node_modules\delayed-stream\test\integration\test-delay
ed-http-upload.js: The system cannot find the path specified.
error for Workspace\PhpStorm\resume-server\registry-server\node_modules\resume-t
o-html\node_modules\resume-schema\node_modules\z-schema\node_modules\request\nod
e_modules\form-data\node_modules\combined-stream\node_modules\delayed-stream\tes
t\integration\test-delayed-stream-auto-pause.js: GetFileAttributesEx D:\Workspac
e\PhpStorm\resume-server\registry-server\node_modules\resume-to-html\node_module
s\resume-schema\node_modules\z-schema\node_modules\request\node_modules\form-dat
a\node_modules\combined-stream\node_modules\delayed-stream\test\integration\test
-delayed-stream-auto-pause.js: The system cannot find the path specified.
error for Workspace\PhpStorm\resume-server\registry-server\node_modules\resume-t
o-html\node_modules\resume-schema\node_modules\z-schema\node_modules\request\nod
e_modules\form-data\node_modules\combined-stream\node_modules\delayed-stream\tes
t\integration\test-delayed-stream.js: GetFileAttributesEx D:\Workspace\PhpStorm\
resume-server\registry-server\node_modules\resume-to-html\node_modules\resume-sc
hema\node_modules\z-schema\node_modules\request\node_modules\form-data\node_modu
les\combined-stream\node_modules\delayed-stream\test\integration\test-delayed-st
ream.js: The system cannot find the path specified.
error for Workspace\PhpStorm\resume-server\registry-server\node_modules\resume-t
o-html\node_modules\resume-schema\node_modules\z-schema\node_modules\request\nod
e_modules\form-data\node_modules\combined-stream\node_modules\delayed-stream\tes
t\integration\test-delayed-stream-pause.js: GetFileAttributesEx D:\Workspace\Php
Storm\resume-server\registry-server\node_modules\resume-to-html\node_modules\res
ume-schema\node_modules\z-schema\node_modules\request\node_modules\form-data\nod
e_modules\combined-stream\node_modules\delayed-stream\test\integration\test-dela
yed-stream-pause.js: The system cannot find the path specified.
error for Workspace\PhpStorm\resume-server\registry-server\node_modules\resume-t
o-html\node_modules\resume-schema\node_modules\z-schema\node_modules\request\nod
e_modules\form-data\node_modules\combined-stream\node_modules\delayed-stream\tes
t\integration\test-handle-source-errors.js: GetFileAttributesEx D:\Workspace\Php
Storm\resume-server\registry-server\node_modules\resume-to-html\node_modules\res
ume-schema\node_modules\z-schema\node_modules\request\node_modules\form-data\nod
e_modules\combined-stream\node_modules\delayed-stream\test\integration\test-hand
le-source-errors.js: The system cannot find the path specified.
error for Workspace\PhpStorm\resume-server\registry-server\node_modules\resume-t
o-html\node_modules\resume-schema\node_modules\z-schema\node_modules\request\nod
e_modules\form-data\node_modules\combined-stream\node_modules\delayed-stream\tes
t\integration\test-max-data-size.js: GetFileAttributesEx D:\Workspace\PhpStorm\r
esume-server\registry-server\node_modules\resume-to-html\node_modules\resume-sch
ema\node_modules\z-schema\node_modules\request\node_modules\form-data\node_modul
es\combined-stream\node_modules\delayed-stream\test\integration\test-max-data-si
ze.js: The system cannot find the path specified.
error for Workspace\PhpStorm\resume-server\registry-server\node_modules\resume-t
o-html\node_modules\resume-schema\node_modules\z-schema\node_modules\request\nod
e_modules\form-data\node_modules\combined-stream\node_modules\delayed-stream\tes
t\integration\test-pipe-resumes.js: GetFileAttributesEx D:\Workspace\PhpStorm\re
sume-server\registry-server\node_modules\resume-to-html\node_modules\resume-sche
ma\node_modules\z-schema\node_modules\request\node_modules\form-data\node_module
s\combined-stream\node_modules\delayed-stream\test\integration\test-pipe-resumes
.js: The system cannot find the path specified.
error for Workspace\PhpStorm\resume-server\registry-server\node_modules\resume-t
o-html\node_modules\resume-schema\node_modules\z-schema\node_modules\request\nod
e_modules\form-data\node_modules\combined-stream\node_modules\delayed-stream\tes
t\integration\test-proxy-readable.js: GetFileAttributesEx D:\Workspace\PhpStorm\
resume-server\registry-server\node_modules\resume-to-html\node_modules\resume-sc
hema\node_modules\z-schema\node_modules\request\node_modules\form-data\node_modu
les\combined-stream\node_modules\delayed-stream\test\integration\test-proxy-read
able.js: The system cannot find the path specified.
←[2K[0:09] 62.27%  54.763 MiB/s  609.473 MiB / 978.800 MiB  19371 / 393... ETA 0
←[2K[0:09] 62.58%  54.763 MiB/s  612.528 MiB / 978.800 MiB  19616 / 393... ETA 0
error for Workspace\PhpStorm\resume-server\registry-server\node_modules\resume-t
o-markdown\node_modules\resume-schema\node_modules\z-schema\node_modules\request
\node_modules\form-data\node_modules\combined-stream\node_modules\delayed-stream
\test\integration\test-delayed-http-upload.js: GetFileAttributesEx D:\Workspace\
PhpStorm\resume-server\registry-server\node_modules\resume-to-markdown\node_modu
les\resume-schema\node_modules\z-schema\node_modules\request\node_modules\form-d
ata\node_modules\combined-stream\node_modules\delayed-stream\test\integration\te
st-delayed-http-upload.js: The system cannot find the path specified.
error for Workspace\PhpStorm\resume-server\registry-server\node_modules\resume-t
o-markdown\node_modules\resume-schema\node_modules\z-schema\node_modules\request
\node_modules\form-data\node_modules\combined-stream\node_modules\delayed-stream
\test\integration\test-delayed-stream-auto-pause.js: GetFileAttributesEx D:\Work
space\PhpStorm\resume-server\registry-server\node_modules\resume-to-markdown\nod
e_modules\resume-schema\node_modules\z-schema\node_modules\request\node_modules\
form-data\node_modules\combined-stream\node_modules\delayed-stream\test\integrat
ion\test-delayed-stream-auto-pause.js: The system cannot find the path specified
.
error for Workspace\PhpStorm\resume-server\registry-server\node_modules\resume-t
o-markdown\node_modules\resume-schema\node_modules\z-schema\node_modules\request
\node_modules\form-data\node_modules\combined-stream\node_modules\delayed-stream
\test\integration\test-delayed-stream-pause.js: GetFileAttributesEx D:\Workspace
\PhpStorm\resume-server\registry-server\node_modules\resume-to-markdown\node_mod
ules\resume-schema\node_modules\z-schema\node_modules\request\node_modules\form-
data\node_modules\combined-stream\node_modules\delayed-stream\test\integration\t
est-delayed-stream-pause.js: The system cannot find the path specified.
error for Workspace\PhpStorm\resume-server\registry-server\node_modules\resume-t
o-markdown\node_modules\resume-schema\node_modules\z-schema\node_modules\request
\node_modules\form-data\node_modules\combined-stream\node_modules\delayed-stream
\test\integration\test-delayed-stream.js: GetFileAttributesEx D:\Workspace\PhpSt
orm\resume-server\registry-server\node_modules\resume-to-markdown\node_modules\r
esume-schema\node_modules\z-schema\node_modules\request\node_modules\form-data\n
ode_modules\combined-stream\node_modules\delayed-stream\test\integration\test-de
layed-stream.js: The system cannot find the path specified.
error for Workspace\PhpStorm\resume-server\registry-server\node_modules\resume-t
o-markdown\node_modules\resume-schema\node_modules\z-schema\node_modules\request
\node_modules\form-data\node_modules\combined-stream\node_modules\delayed-stream
\test\integration\test-handle-source-errors.js: GetFileAttributesEx D:\Workspace
\PhpStorm\resume-server\registry-server\node_modules\resume-to-markdown\node_mod
ules\resume-schema\node_modules\z-schema\node_modules\request\node_modules\form-
data\node_modules\combined-stream\node_modules\delayed-stream\test\integration\t
est-handle-source-errors.js: The system cannot find the path specified.
error for Workspace\PhpStorm\resume-server\registry-server\node_modules\resume-t
o-markdown\node_modules\resume-schema\node_modules\z-schema\node_modules\request
\node_modules\form-data\node_modules\combined-stream\node_modules\delayed-stream
\test\integration\test-max-data-size.js: GetFileAttributesEx D:\Workspace\PhpSto
rm\resume-server\registry-server\node_modules\resume-to-markdown\node_modules\re
sume-schema\node_modules\z-schema\node_modules\request\node_modules\form-data\no
de_modules\combined-stream\node_modules\delayed-stream\test\integration\test-max
-data-size.js: The system cannot find the path specified.
error for Workspace\PhpStorm\resume-server\registry-server\node_modules\resume-t
o-markdown\node_modules\resume-schema\node_modules\z-schema\node_modules\request
\node_modules\form-data\node_modules\combined-stream\node_modules\delayed-stream
\test\integration\test-pipe-resumes.js: GetFileAttributesEx D:\Workspace\PhpStor
m\resume-server\registry-server\node_modules\resume-to-markdown\node_modules\res
ume-schema\node_modules\z-schema\node_modules\request\node_modules\form-data\nod
e_modules\combined-stream\node_modules\delayed-stream\test\integration\test-pipe
-resumes.js: The system cannot find the path specified.
error for Workspace\PhpStorm\resume-server\registry-server\node_modules\resume-t
o-markdown\node_modules\resume-schema\node_modules\z-schema\node_modules\request
\node_modules\form-data\node_modules\combined-stream\node_modules\delayed-stream
\test\integration\test-proxy-readable.js: GetFileAttributesEx D:\Workspace\PhpSt
orm\resume-server\registry-server\node_modules\resume-to-markdown\node_modules\r
esume-schema\node_modules\z-schema\node_modules\request\node_modules\form-data\n
ode_modules\combined-stream\node_modules\delayed-stream\test\integration\test-pr
oxy-readable.js: The system cannot find the path specified.
```

But when i tried to restore from snapshots, it wasn't able to and give following error
```
D:\Workspace\Go-Ide\GOPATH\src\restic>restic.exe -r D:/Restic-Backups restore ab
ae29a7 --target D:/Restic-Restore
enter password for repository:
restoring <Snapshot of [D:\Workspace] at 2015-11-25 19:19:59.6134143 +0500 PKT>
to D:/Restic-Restore
error for Workspace\APKStudio: restore subtree: UtimesNano: Access is denied.
error for Workspace: restore subtree: UtimesNano: Access is denied.
UtimesNano: Access is denied.

D:\Workspace\Go-Ide\GOPATH\src\restic>restic.exe -r D:/Restic-Backups restore ab
ae29a7 --target D:/Restic-Restore
enter password for repository:
restoring <Snapshot of [D:\Workspace] at 2015-11-25 19:19:59.6134143 +0500 PKT>
to D:/Restic-Restore
error for D:\Restic-Restore\Workspace: create node: createDirAt: Mkdir: mkdir D:
\Restic-Restore\Workspace: Cannot create a file when that file already exists.
error for D:\Restic-Restore\Workspace\APKStudio: create node: createDirAt: Mkdir
: mkdir D:\Restic-Restore\Workspace\APKStudio: Cannot create a file when that fi
le already exists.
error for D:\Restic-Restore\Workspace\APKStudio\TextraAPK: create node: createDi
rAt: Mkdir: mkdir D:\Restic-Restore\Workspace\APKStudio\TextraAPK: Cannot create
 a file when that file already exists.
error for Workspace\APKStudio: restore subtree: UtimesNano: Access is denied.
error for Workspace: restore subtree: UtimesNano: Access is denied.
UtimesNano: Access is denied.

D:\Workspace\Go-Ide\GOPATH\src\restic>restic.exe -r D:/Restic-Backups restore ab
ae29a7 --target D:/Restic-Restore
enter password for repository:
restoring <Snapshot of [D:\Workspace] at 2015-11-25 19:19:59.6134143 +0500 PKT>
to D:/Restic-Restore
error for Workspace\APKStudio: restore subtree: UtimesNano: Access is denied.
error for Workspace: restore subtree: UtimesNano: Access is denied.
UtimesNano: Access is denied.

D:\Workspace\Go-Ide\GOPATH\src\restic>restic.exe -r D:/Restic-Backups restore 90
aa8088 --target D:/Restic-Restore
enter password for repository:
restoring <Snapshot of [D:\Workspace] at 2015-11-25 19:04:44.6940839 +0500 PKT>
to D:/Restic-Restore
error for Workspace\APKStudio: restore subtree: UtimesNano: Access is denied.
error for Workspace: restore subtree: UtimesNano: Access is denied.
UtimesNano: Access is denied.
```

is there anything i am doing wrong? I will really appreciate your help.

Thanks

**Comment by fd0** [comment]
I don't have the time today to look into this in detail, but could you try again with backslash-characters in the paths? e.g. `restic.exe -r D:\Restic-Backups restore 90
aa8088 --target D:\Restic-Restore`

**Comment by tigerclawn82** [comment]
no problem, i was just evaluating scalable solution for backups. I tried with backslash but still same error.

**Comment by fd0** [comment]
Did restic restore anything? From the messages it seems that it was only unable to restore the exact timestamp for the directories, but it should've restored the files...

**Comment by tigerclawn82** [comment]
No, it didn't restore anything. I tried several times but no luck. I tried to backup another dir without any binary files, there were no errors on backup but i still wasn't able to restore from snapshot.

**Comment by rueger** [comment]
I'm not a windows expert, but i think you hit Windows [Maximum Path Length Limitation](https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247(v=vs.85).aspx#maxpath). There you can also find how to extend the 260 char limit to 32k.

**Comment by fd0** [comment]
This is fixed in the latest master branch since #570.
